### DISCUSSION TITLE
Can't find source to System.Net.WebSockets.BrowserWebSocket anywhere

### DISCUSSION THREAD

**Discussion by KieranDevvs** [dotnet/runtime#74522] [Q&A] [answered]
I want to see how the websocket frame header is constructed (if at all) under Blazor WASM. The `ClientWebSocket` implementation instantiates `System.Net.WebSockets.BrowserWebSocket` 
![image](https://user-images.githubusercontent.com/21192520/186522033-c74c4542-0a29-4cf3-ab15-b8e250eafb11.png)
https://source.dot.net doesn't contain a reference to this type, nor can I find anything on google.

What's going on?

**Comment by PathogenDavid** [comment, marked as answer]
It lives here: https://github.com/dotnet/runtime/blob/24d185e53ad52319799f345f38c5c4fb6377e3a0/src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

https://source.dot.net/ has been [having some issues lately](https://github.com/dotnet/source-indexer/issues/103), so this might be related.

I found it using [GitHub Code Search](https://cs.github.com/?scopeName=All+repos&scope=&q=%22class+BrowserWebSocket%22+language%3AC%23+org%3Adotnet) instead. If you aren't in the Code Search preview, you can also usually find things by going to the runtime repo, pressing <kbd>T</kbd> to open the file finder and typing the class name.

**Reply by KieranDevvs** [reply]
Thanks
